concourse/pipelines/sap-workload-test.yaml

```diff
@@ -0,0 +1,149 @@
+jobs:
+- name: sap-workload-test
+  plan:
+  - task: generate-unique-id
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: busybox
+      run:
+        path: sh
+        args:
+        - -exc
+        - |
+          # TODO - make more unique? guid?
+          echo $RANDOM | tee unique-id/id
+      outputs:
+      - name: unique-id
+
+  - load_var: id
+    file: unique-id/id
+
+  - task: generate-post-script
+    config:
+      platform: linux
+      image_resource:
+        type: docker-image
+        source:
+          repository: google/cloud-sdk
+          tag: latest
+      run:
+        path: sh
+        args:
+        - -exc
+        - |
+          cat <<EOT >>sap_post_script.sh
+          if cat /var/log/messages | grep -q "ERROR - Deployment Exited"; then
+            echo "ERROR" > result.txt
+          else
+            echo "SUCCESS" > result.txt
+          fi
+          gsutil cp result.txt gs://test-bucket-for-terraform/((.:id))/run_result
+          EOT
+
+          gsutil cp sap_post_script.sh gs://test-bucket-for-terraform/((.:id))/sap_post_script.sh
+
+  - task: create-sap-tf-environment
+    config:
+      platform: linux
+      image_resource:
+        type: docker-image
+        source:
+          repository: hashicorp/terraform
+          tag: latest
+      run:
+        path: sh
+        args:
+        - -exc
+        - |
+          cat <<EOT >> sap_hana.tf
+          terraform {
+            backend "gcs" {
+              bucket = "test-bucket-for-terraform"
+              prefix = "((.:id))/terraform"
+            }
+          }
+          module "sap_hana" {
```

Comment:
@hopkiw You mentioned if we could move this chunk of code into its own file.  I'm keeping it here because we'll need to set some of the variables during script execution (like the instance_name, post_deployment_script and image information).

Do you think it's still appropriate to have a separate file?

Reply:
we could have flat, checked-in files. and parameterize them with `-var:` arguments at runtime. i think that is a better model than generating the file dynamically, wdyt?